include/deal.II/base/quadrature_point_data.h

```diff
@@ -153,23 +160,51 @@ class CellDataStorage : public Subscriptor
    * cell-by-cell basis.
    *
    * @pre The type @p T needs to match the class provided to initialize() .
+   *
+   * @pre @p cell must be from the same triangulation that is used to
+   * initialize() the cell data.
    */
   template <typename T = DataType>
   std::vector<std::shared_ptr<const T>>
   get_data(const CellIteratorType &cell) const;
 
 private:
   /**
-   * A map to store a vector of data on a cell.
+   * Depending on @p CellIteratorType we need to define a triangulation type
+   */
+  using triangulation_type =
+    Triangulation<CellIteratorType::AccessorType::dimension,
+                  CellIteratorType::AccessorType::space_dimension>;
```

Comment:
The normal way we do that is to use
```
static const unsigned int dimension = CellIteratorType::AccessorType::dimension;
```
etc. Could you do that for `dimension` and `space_dimension` and then remove this using statement? They do the same thing, but we usually save the spatial dimensions in this way.